src/shadowbox/server/manager_service.spec.ts

```diff
@@ -738,6 +740,79 @@ describe('ShadowsocksManagerService', () => {
   });
 });
 
+describe('bindService', () => {
+  let server: restify.Server;
+  let service: ShadowsocksManagerService;
+  let url: URL;
+  const prefix = '/TestApiPrefix';
+
+  const fakeResponse = {'foo': 'bar'};
+  const fakeHandler = async (req, res, next) => {
+    res.send(200, fakeResponse);
+    next();
+  };
+
+  beforeEach(() => {
+    server = restify.createServer();
+    service = new ShadowsocksManagerServiceBuilder().build();
+    server.listen(0);
+    url = new URL(server.url);
+  });
+
+  afterEach(() => {
+    server.close();
+  });
+
+  it('basic routing', async () => {
+    spyOn(service, "renameServer").and.callFake(fakeHandler);
+    bindService(server, prefix, service);
+
+    url.pathname = `${prefix}/name`;
+    const response = await fetch(url, {method: 'put'});
+    const body = await response.json();
+
+    expect(body).toEqual(fakeResponse);
+    expect(service.renameServer).toHaveBeenCalled();
+  });
+
+  it('parameterized routing', async () => {
+    spyOn(service, "removeAccessKeyDataLimit").and.callFake(fakeHandler);
+    bindService(server, prefix, service);
+
+    url.pathname = `${prefix}/access-keys/fake-access-key-id/data-limit`;
+    const response = await fetch(url, {method: 'delete'});
+    const body = await response.json();
+
+    expect(body).toEqual(fakeResponse);
+    expect(service.removeAccessKeyDataLimit).toHaveBeenCalled();
+  });
+
+  // Verify that we have consistent 404 behavior for all inputs.
+  [
+    '/',
+    '/TestApiPre',
+    '/foo',
```

Comment:
Perhaps add `/TestApiPrefixxx` to make sure it's an exact match?

Reply:
Done